src/puppetlabs/trapperkeeper/services/status/status_core.clj

```diff
@@ -264,20 +264,64 @@
                :message (str "Invalid service_status_version. Should be an "
                           "integer but was " level)}))))
 
+(schema/defn ^:always-validate summarize-states :- State
+  "Given a map of service statuses:
+   * if all of the statuses have the same :state, returns that :state
+   * if not all of the statuses are the same, returns :unknown"
+  [statuses :- ServicesStatus]
+  (let [state-set (->> statuses
+                       vals
+                       (map :state)
+                       set)]
+    (cond
+      (state-set :error) :error
+      (state-set :unknown) :unknown
+      (state-set :running) :running
+      :else :runing)))
```

Comment:
So first off there's a type here, but it seems safest to report :unknown if we get here (given that we don't think it should even be possible).